src/it/resources/expected/cppcli_circular_dependent_interface/cppcli/AnotherInterface.hpp

```diff
@@ -0,0 +1,30 @@
+// AUTOGENERATED FILE - DO NOT MODIFY!
+// This file was generated by Djinni from cppcli_circular_dependent_interface.djinni
+
+#pragma once
+
+#include "../cpp-headers/another_interface.hpp"
+#include "OneInterface.hpp"
+#include <memory>
+
+ref class OneInterface;
```

Comment:
I'm confused, I don't get how just the forward-declaration of the type alone resolves the problem with the circular dependency. Afaik the dependency resolution in this case should still face be the same recursive problem?

`#include "OneInterface.hpp"` -> contains `#include "AnotherInterface.hpp"` -> contains `#include "OneInterface.hpp"`, ...

As far as I understand it to break the circle we would really have to detect that there is a circle and not include the other file in one of the headers? Correct me if I'm wrong though! 🤔

Reply:
I guess `#pragma once` in each header does the trick. It seems there are many discussions, e.g. http://www.cplusplus.com/forum/general/243162/.